Feat/place system - 상세조회 기능 구현/사용자 북마크 조회기능

balsohn opened the pull request:
## ✅ 작업 개요 (What)
상세조회 기능 구현
응답값에 사진 추가

## 🎯 작업 목적 (Why)
placeId만 가지고 장소의 정보를 가져오는 기능

## 🛠️ 작업 내용 (How)
사용자의 북마크 목록을 조회할때 출력을 하기위해

## ⚠️ 고려 사항


## 📎 참고 자료

balsohn committed:
feat: 사용자 북마크 조회 및 장소 상세 정보 기능 완성

- Google Place Details API 완전 구현
- 사용자 북마크 목록 조회 기능 완성
- 장소 상세 조회 API 엔드포인트 추가

balsohn committed:
feat: 북마크 관리 완성 및 장소 사진 기능 추가

- 북마크 삭제 API 구현
- 장소 대표 사진 추가
- HTTP 테스트 케이스 추가

balsohn committed:
fix: 충돌 해결

hyohee0613 commented on src/main/java/com/example/wherewego/domain/places/controller/PlaceController.java:
```diff
@@ -104,10 +106,36 @@ public ResponseEntity<ApiResponse<BookmarkCreateResponseDto>> addBookmark(
 	 * 사용자 북마크 목록 조회 API
 	 *
-	 * GET /api/users/bookmarks
+	 * GET /api/places/users/bookmarks
```
내가 누른 장소 북마크 목록 조회가 마이페이지에 있는 거니까 url을 바꿔야하지 않을까 싶습니다 
/api/users/mypage/placebookmark  이런식으루?

balsohn replied:
그래서 맨 처음에 user쪽에서 맡았던 부분인데 제가 가져와서 하다보니 api가 애매해진거같습니다.
제쪽컨트롤러에서 작성하고 책임도 제쪽에 있다보니 이렇게 작성한 부분인데
api 그렇게 수정을 바라면 제가  수정하기보다는 유저쪽에서 다시 작업하시는게 맞다고 봅니다.
일관성은 사라지긴 하지만, 충분히 RESTFul 하다 생각해서 작성했습니다.

balsohn replied:
유저 컨트롤러로 이동시켰습니다

balsohn committed:
feat: Course 도메인용 경로 거리 계산 API 구현

- CoursePlaceInfo DTO
  - distanceFromUser: 사용자 위치로부터 직선거리
  - distanceFromPrevious: 이전 장소로부터의 이동거리
  - visitOrder: 코스 내 방문 순서

- CourseRouteSummary DTO(선택사항)
  - totalDistance: 총 경로 거리
  - placeCount: 총 장소 개수

balsohn committed:
Merge branch 'dev' of https://github.com/two2gether/where-we-go into feat/place-system

balsohn committed:
refactor: 북마크 조회 엔드포인트를 마이페이지로 이동

- getUserBookmarks -> getMyBookmarks
- ㅔãPlace 패키지 -> User 패키지
